### PR TITLE
libc/lib_memcpy.c:Add mempcpy method.

### PR DESCRIPTION
**Pull request by crafcat7**
## Summary
libc/lib_memcpy.c:Add mempcpy method.

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: libs/libc/string/lib_mempcpy.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: mempcpy
+ ****************************************************************************/
+
+FAR void *mempcpy(FAR void *dest, FAR const void *src, size_t n)
+{
+  return memcpy(dest, src, n) + n;
```
[the remark this answers is not in the document]
Comment (hartmannathan) [reply libs/libc/string/lib_mempcpy.c:39]: Agreed; I think that on some compilers void pointer increment will cause (at least) a warning. Maybe an error.

[the remark this answers is not in the document]
Comment (hartmannathan) [reply libs/libc/string/lib_mempcpy.c:39]: You could try something like:

```
return (void *)((char *) memcpy(dest, src, n) + n);
```

I know, it's ugly. :-/

File: libs/libc/string/lib_mempcpy.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#include <sys/types.h>
+#include <string.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: mempcpy
+ ****************************************************************************/
```
Comment (hartmannathan) [inline libs/libc/string/lib_mempcpy.c:35]: I recommend a docstring. For example (feel free to commit as-is):

```suggestion
 *
 * Description:
 *   Like memcpy(), but returns address of byte after the last copied byte
 *   in dest. This allows constructing data piecewise by copying its parts
 *   into consecutive memory locations.
 *
 *   This function is not in POSIX.
 *
 * Input Parameters:
 *   src  - Source location from which to copy.
 *   dest - Destination location into which to copy.
 *   n    - Size of data to be copied, in bytes.
 *
 * Returned Value:
 *   A pointer to the byte in dest which immediately follows the last copied
 *   byte.
 *
 ****************************************************************************/
```

File: libs/libc/string/lib_mempcpy.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkarashchenko) [reply libs/libc/string/lib_mempcpy.c:39]: `return (char *)memcpy(dest, src, n) + n;` is enough

File: libs/libc/string/lib_mempcpy.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * Returned Value:
+ *   A pointer to the byte in dest which immediately follows the last copied
+ *   byte.
+ *
+ ****************************************************************************/
+
+FAR void *mempcpy(FAR void *dest, FAR const void *src, size_t n)
+{
+  return (char *)memcpy(dest, src, n) + n;
```
Comment (pkarashchenko) [inline libs/libc/string/lib_mempcpy.c:60]:
```suggestion
  return (FAR char *)memcpy(dest, src, n) + n;
```

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@crafcat7 sqaush your patch.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**